tests/index.test.ts

```diff
@@ -260,7 +260,7 @@ test(`won't parse body if Content-Length header doesn't exist`, async t => {
 
 test(`body does not exist, will not fail to parse non-json`, async t => {
   const server = createServer((req, res) => {
-    res.writeHead(200);
+    res.writeHead(204);
```

Comment:
What happens without this change? I think it's important to make sure we understand what happens when the body is empty but someone tries to run `.json()` anyway.

Reply:
If the body is empty, we _should_ get a 204 No Content response from the API.

If we do not get a 204, and instead get a 200, we'd need to add more logic for that. I wasn't sure if that was a good idea or not.

Reply:
I think the behavior of this code is correct, but I think it's important to keep the test for "what happens when we get no content and an HTTP 200".

Reply:
Fixed - we read the response as text, then only try to parse to JSON if there is length. If there is any content and it's invalid JSON, we get an error. If there is no content, we just get an empty 200 response.

Reply:
Won't this also change behavior for other methods? For example, `.text()` would previously return `undefined` when an HTTP 200 has no content, but now it will return `""`.

I want to reiterate: I think it's 100% fine to just remove the "is there content length" check and let the chips fall where they may -- I'm worried that making a specific cutout for `.json()` is adding complexity where we don't need it.

Happy to pair if you'd like, I appreciate you taking care of this!

Reply:
Tests break if we just remove the content length check - there is a specific test to check if JSON is parsed when there is no content.

We should still get `undefined` when `.text()` is run on an empty response. Empty string evaluates as falsy.

Reply:
Yep, changing the test outcomes sounds correct to me, mostly I want to make sure that we aren't just changing the test _setups_ rather than the outcomes. In other words: I'd like to keep the "given when" and replace the "then". 😄 


> We should still get undefined when .text() is run on an empty response. Empty string evaluates as falsy.

I agree that we should! But the code, as-written, looks like this:

```javascript
if (status !== 204) {
          if (key === "json") {
            const text = await response.clone().text();
            if (text) {
              const parsed = JSON.parse(text);
              resolved = parsed;
            }
          } else {
            resolved = await response.clone()[key]();
          }
        }
```

****

So `resolved = response.clone().text()`, which is `""`. We never hit the `if (text)` because `"text" == "json"` evaluates to false.

Reply:
Ah shoot - I see what you're saying there (specifically when `.text()` is the method instead of `.json()`). There is no test to guarantee that if the response is blank that `.text()` should return `undefined` instead of `""` so we may be ok. I can force it to `undefined`, but I'm not sure if `""` results in a Content-Length of 0 or a number greater than zero.